doc/decisions/frontend_documentation.md

```diff
@@ -0,0 +1,96 @@
+# Frontend Documentation
+
+## Problem
+
+We need to choose a workflow, tooling that helps us to keep the documentation up to date.
+An easy to read and up to date documentation reduces the entry barrier for new-comers.
+Without proper documentation maintenance becomes increasingly more difficult as the project grows.
+
+## Constraints
+
+1. The documentation should be easy to navigate.
+2. The documentation should for the most part, be automatically extracted from the source code.
+
+## Assumptions
+
+1. With a good and beautiful documentation tool, developers are more likely to use and maintain it.
+2. With a documentation tool that enables the viewing of react components, the code quality is improved. Presentation components, should not contain logic with side effects.
+Hooks should be used outside of presentation components, with props passed into them.
```

Comment:
This should be in general guidelines about frontend (architecture), and we should like from here to this document.

Reply:
@markus2330 I added it to `contrib/frontend`.